### ISSUE TITLE
Unknown error occurred when configuring with the UI.

### ISSUE THREAD

**Issue by sytone** [eelcohn/home-assistant-ddwrt#19]
Caused by an issue with HA seeing it as a blocking call.

```
2023-01-09 14:19:01.976 DEBUG (MainThread) [custom_components.ddwrt.pyddwrt] DDWrt.update_about_data: Updating about data...
2023-01-09 14:19:01.976 DEBUG (MainThread) [custom_components.ddwrt.pyddwrt] DDWrt._get_ddwrt_data: Connecting to http://192.168.0.1/About.htm
2023-01-09 14:19:01.986 WARNING (MainThread) [homeassistant.util.async_] Detected blocking call to putrequest inside the event loop. This is causing stability issues. Please report issue to the custom integration author for ddwrt doing blocking calls at custom_components/ddwrt/pyddwrt.py, line 860: response = self._session.get(
2023-01-09 14:19:01.987 DEBUG (MainThread) [custom_components.ddwrt.pyddwrt] DDWrt._get_ddwrt_data: Unable to connect to the router Connection error: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; at custom_components/ddwrt/pyddwrt.py, line 860: response = self._session.get(
2023-01-09 14:19:01.988 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/config/custom_components/ddwrt/pyddwrt.py", line 860, in _get_ddwrt_data
    response = self._session.get(
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 600, in get
    return self.request("GET", url, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 489, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 398, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.10/site-packages/urllib3/connection.py", line 239, in request
    super(HTTPConnection, self).request(method, url, body=body, headers=headers)
  File "/usr/local/lib/python3.10/http/client.py", line 1282, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.10/http/client.py", line 1293, in _send_request
    self.putrequest(method, url, **skips)
  File "/usr/local/lib/python3.10/site-packages/urllib3/connection.py", line 219, in putrequest
    return _HTTPConnection.putrequest(self, method, url, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 180, in protected_loop_func
    check_loop(func, strict=strict)
  File "/usr/src/homeassistant/homeassistant/util/async_.py", line 168, in check_loop
    raise RuntimeError(
RuntimeError: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; at custom_components/ddwrt/pyddwrt.py, line 860: response = self._session.get(

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/ddwrt/pyddwrt.py", line 142, in update_about_data
    self.data = self._get_ddwrt_data(url, False)
  File "/config/custom_components/ddwrt/pyddwrt.py", line 909, in _get_ddwrt_data
    raise(DDWrt.ExceptionUnknown(e))
custom_components.ddwrt.pyddwrt.DDWrt.ExceptionUnknown: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; at custom_components/ddwrt/pyddwrt.py, line 860: response = self._session.get(

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/config/custom_components/ddwrt/config_flow.py", line 154, in async_step_user
    valid_router = router.update_about_data()
  File "/config/custom_components/ddwrt/pyddwrt.py", line 145, in update_about_data
    raise(DDWrt.DDWrtException("Unable to update about data: %s", e))
custom_components.ddwrt.pyddwrt.DDWrt.DDWrtException: ('Unable to update about data: %s', ExceptionUnknown(RuntimeError('Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; at custom_components/ddwrt/pyddwrt.py, line 860: response = self._session.get(')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 222, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 81, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 136, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 180, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 73, in wrapper
    result = await method(view, request, data, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 281, in async_configure
    result = await self._async_handle_step(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 368, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/ddwrt/config_flow.py", line 160, in async_step_user
    except CannotConnect:
NameError: name 'CannotConnect' is not defined
```

**Comment by olegglarin** [comment]
Facing the same issue under Home Assistant 2022.9.2

**Comment by homeautomaton** [comment]
I am investigating this problem.

It's a bit puzzling, because I find the appropriate call to async_add_executor job here:

```
/config/custom_components/ddwrt/__init__.py:            result = await self._hass.async_add_executor_job(self._router.update_lan_data) 
```

EDIT: I found a synchronous call, elsewhere, in config_flow.py. On it.

**Comment by eelcohn** [comment]
Should be fixed in v1.1.4. Thanks to @homeautomaton for investigating this issue!
